types/defines/images.d.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+type InfoResponse =
+  | { format: "image/svg+xml" }
+  | {
+      format: string;
+      fileSize: number;
+      width: number;
+      height: number;
+    };
+
+type RawInfoResponse =
+  | { format: "image/svg+xml" }
+  | {
+      format: string;
+      file_size: number;
+      width: number;
+      height: number;
+    };
+
+type Transform = {
+  fit?: "scale-down" | "contain" | "pad" | "squeeze" | "cover" | "crop";
+  gravity?:
+    | "left"
+    | "right"
+    | "top"
+    | "bottom"
+    | "center"
+    | "auto"
+    | "entropy"
+    | "face"
+    | {
+        x?: number;
+        y?: number;
+        mode: "remainder" | "box-center";
+      };
+  trim?: {
+    top?: number;
+    bottom?: number;
+    left?: number;
+    right?: number;
+    width?: number;
+    height?: number;
+    border?:
+      | boolean
+      | {
+          color?: string;
+          tolerance?: number;
+          keep?: number;
+        };
+  };
+  width?: number;
+  height?: number;
+  background?: string;
+  rotate?: number;
+  sharpen?: number;
+  blur?: number;
+  contrast?: number;
+  brightness?: number;
+  gamma?: number;
+  border?: {
+    color?: string;
+    width?: number;
+    top?: number;
+    bottom?: number;
+    left?: number;
+    right?: number;
+  };
+  zoom?: number;
+};
+
+type OutputOptions = {
+  format:
+    | "image/jpeg"
+    | "image/png"
+    | "image/gif"
+    | "image/webp"
+    | "image/avif"
+    | "rgb"
+    | "rgba";
+  quality?: number;
+  background?: string;
+};
+
+interface ImagesBinding {
+  info(blob: Blob): Promise<InfoResponse>;
+  input(blob: Blob): ImageTransformer;
```

Comment:
could the input be a stream instead? It's easy to turn a blob to a stream, the opposite is problematic.

Reply:
my main goal here is to enable streaming transformations/use-cases by default, because enabling that ad-hoc will be hard/awkward.

Reply:
I've made the interfaces take streams and then buffer internally for now as it's a bit tricky to stream properly with multipart, which is what the backend uses